app/[locale]/(authenticated)/profile/actions.ts

```diff
@@ -0,0 +1,68 @@
+"use server";
+
+import * as z from "zod";
+import { query } from "@/app/db";
+import { Scrypt } from "oslo/password";
+import { getTranslations } from "next-intl/server";
+import { FormState } from "@/app/components/Form";
+
+const scrypt = new Scrypt();
+
+const profileValidationSchema = z
+  .object({
+    email: z.string().min(1),
+    name: z.string().min(1),
+    password: z.union([z.string().min(8), z.literal("")]),
+    confirmPassword: z.string(),
```

Comment:
We need to make these two optional. If you just want to change your name, you shouldn't have to update your password too